server/src/main/java/org/elasticsearch/index/store/Store.java

```diff
@@ -446,13 +446,8 @@ public static void tryOpenIndex(Path indexLocation, ShardId shardId, NodeEnviron
         try (ShardLock lock = shardLocker.lock(shardId, "open index", TimeUnit.SECONDS.toMillis(5));
              Directory dir = new SimpleFSDirectory(indexLocation)) {
             failIfCorrupted(dir);
-            // Previously we called Lucene#readSegmentInfos which verifies that some Lucene metadata is readable and makes sense, but if it
-            // weren't then we would mark this shard as corrupt when allocated, so it seems that this is unnecessary (and it breaks when
-            // the shard's directory is virtual since we use SimpleFSDirectory above.
-            // TODO NORELEASE is this ok? Need to check that we definitely add a corruption marker if the metadata is corrupt.
```

Comment:
With this commit, this hack is no longer needed since we do not fetch the shard info like this.

Reply:
👍